hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmKeyInfo.java

```diff
@@ -210,6 +221,25 @@ public void setKeyLocationVersions(
     this.keyLocationVersions = keyLocationVersions;
   }
 
+  private void refreshCurrentLocationPartsMap() {
+    if (this.keyLocationVersions.size() > 0) {
+      this.currentlocationsPartsMap = this.keyLocationVersions.get(keyLocationVersions.size() - 1)
+              .getLocationList()
+              .stream()
+              .filter(it -> it.getPartNumber() != 0)
+              .collect(Collectors.groupingBy(OmKeyLocationInfo::getPartNumber));
+      this.currentDataSizePartsMap = this.currentlocationsPartsMap.entrySet().stream()
+              .map(it -> Pair.of(
+                      it.getKey(),
+                      it.getValue().stream().
+                              mapToLong(BlockLocationInfo::getLength)
+                              .sum()
+                      )
+              )
+              .collect(Collectors.toMap(Pair::getKey, Pair::getValue));
+    }
+  }
+
```

Comment:
I think to just move this logic to `KeyManagerImpl`.

Reply:
Done.